### PR TITLE
docs: user documentation for NFS

### PR DESCRIPTION
**Pull request by tmilos77**
**Description**

Changes proposed in this pull request:

- user documentation for NFS

**Related issue(s)**

### CODE REVIEW

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+   awsnfsvolume.cloud-resources.kyma-project.io/my-vol condition met
+   ```
+   
+4. Observe the generated PersistentVolume:
+   
+   ```shell
+   kubectl -n $NAMESPACE get persistentvolume my-vol
+   ```
+   
+   This command should print:
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:45]:
```suggestion
   You should see the following details:
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+   
+   ```
+   NAME     CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM            STORAGECLASS
+   my-vol   100G       RWX            Retain           Bound    test-mt/my-vol               
+   ```
+   
+   Note the RWX access mode which allows volume to be readable and writable from multiple workloads, and
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:52]:
```suggestion
   Note the `RWX` access mode which allows the volume to be readable and writable from multiple workloads, and the
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+   Bound status which means PersistentVolumeClaim claiming this PV is created.
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:53]:
```suggestion
   the `Bound` status which means the PersistentVolumeClaim claiming this PV is created.
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+   
+5. Observe the generated PersistentVolumeClaim:
+
+   ```shell
+   kubectl -n $NAMESPACE get persistentvolumeclaim my-vol
+   ```
+
+   This command should print:
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:61]:
```suggestion
   You should see the following message:
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+   ```
+   NAME     STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS 
+   my-vol   Bound    my-vol   100G       RWX                         
+   ```
+
+   Similarly to PV, note the RWX access mode and Bound status.
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:68]:
```suggestion
   Similarly to PV, note the `RWX` access mode and `Bound` status.
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+6. Create two workloads that both will write to the volume
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:70]:
```suggestion
6. Create two workloads that both write to the volume
```
What does it mean that the workloads write to the volume?

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+             configMap:
+               name: my-script
+               defaultMode: 0744 
+           - name: data
+             persistentVolumeClaim:
+               claimName: my-vol 
+   EOF
+   ``` 
+   
+   This workload will print a sequence of 20 lines to stdout and a file on the nfs volume.
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:131]:
```suggestion
   This workload should print a sequence of 20 lines to stdout and a file on the NFS volume.
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+   Then it will print the content of the file.
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:132]:
```suggestion
   Then it should print the content of the file.
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+   
+7. Print the logs of one of the workloads
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:134]:
```suggestion
7. Print the logs of one of the workloads, run:
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+   Hello from awsnfsvolume-demo-869c89df4c-l8hrb: 2
+   Hello from awsnfsvolume-demo-869c89df4c-dsw97: 2
+   Hello from awsnfsvolume-demo-869c89df4c-l8hrb: 3
+   ...
+   ```
+   
+   Note that the content after `File content:` contains prints from both workloads. This 
+   demonstrates the ReadWriteMany capability of the volume.
+
+8. Clean up
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:158]: The clean-up steps can be formatted as an unordered list.

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+   Remove the created awsnfsvolume:
+   ```shell
+   kubectl delete -n $NAMESPACE awsnfsvolume my-vol
+   ```
+
+   Remove the created default iprange:
+
+   > [!NOTE]
+   > If you have other cloud resources using the default IpRange,
+   > then you should skip this step, and not delete the default IpRange.
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:179]:
```suggestion
   > skip this step, and do not delete the default IpRange.
```

File: docs/user/tutorials/README.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+# Tutorials
+
+Browse the Cloud Manager tutorials to learn how to create and use cloud resources.
+
+* [AwsNfsVolume](./01-10-aws-nfs-volume.md)
```
Comment (mmitoraj) [inline docs/user/tutorials/README.md:5]: If you change the title as suggested, please also update it here: "Use RWX Volumes in AWS"

**Review by mmitoraj — COMMENTED** [review]
- I will create a follow-up task to update the diagrams as per our [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/09-diagrams.md).
- The main README in the repo requires updating, we can create a follow-up for that as well.
- I'm thinking about creating a glossary document for all the terms, such as AWS EFS, RWX, CIDR. Maybe we could explain those abbreviations in one doc and provide some basic details about the terms?

File: docs/user/tutorials/01-10-aws-nfs-volume.md (hunk 6 above)
Comment (tmilos77) [reply docs/user/tutorials/01-10-aws-nfs-volume.md:70]: Both workloads writing to volume means there are two pods with apps writing to the same file on that NFS disk / volume

File: docs/user/README.md [hunk 13]
```diff
@@ -1,14 +1,45 @@
...
 
-# {Module Name}
-> Modify the title and insert the name of your module. Use Heading 1 (H1).
+# Cloud Manager Module
 
-## Overview
-> Provide a description of your module and its components. Describe its features and functionalities. Mention the scope and add information on the CustomResourceDefinitions (CRDs).
-> You can divide this section to the relevant subsections.
+## What is Cloud Manager Module?
+Cloud Manager is a central component that manages access to additional Hyperscaler resources from the
```
Comment (mmitoraj) [inline docs/user/README.md:5]:
```suggestion
Cloud Manager is a central component that manages access to additional hyperscaler resources from the 
```

File: docs/user/README.md [hunk 14]
```diff
@@ -1,14 +1,45 @@
...
-# {Module Name}
-> Modify the title and insert the name of your module. Use Heading 1 (H1).
+# Cloud Manager Module
 
-## Overview
-> Provide a description of your module and its components. Describe its features and functionalities. Mention the scope and add information on the CustomResourceDefinitions (CRDs).
-> You can divide this section to the relevant subsections.
+## What is Cloud Manager Module?
+Cloud Manager is a central component that manages access to additional Hyperscaler resources from the 
+Kyma Runtime cluster. Its responsibility is to bring Hyperscaler products/resources onto the Kyma cluster
```
Comment (mmitoraj) [inline docs/user/README.md:6]:
```suggestion
Kyma Runtime cluster. Its responsibility is to bring hyperscaler products/resources into the Kyma cluster 
```

File: docs/user/README.md [hunk 15]
```diff
@@ -1,14 +1,45 @@
...
+# Cloud Manager Module
 
-## Overview
-> Provide a description of your module and its components. Describe its features and functionalities. Mention the scope and add information on the CustomResourceDefinitions (CRDs).
-> You can divide this section to the relevant subsections.
+## What is Cloud Manager Module?
+Cloud Manager is a central component that manages access to additional Hyperscaler resources from the 
+Kyma Runtime cluster. Its responsibility is to bring Hyperscaler products/resources onto the Kyma cluster 
+in a secure way. Once Cloud Manager as a module is enabled in the Kyma cluster, Cloud Manager's features 
+will give you access to the respective products and resources of the Hyperscaler providers.
```
Comment (mmitoraj) [inline docs/user/README.md:8]:
```suggestion
give you access to the respective products and resources of the hyperscaler providers.
```

File: docs/user/README.md [hunk 16]
```diff
@@ -1,14 +1,45 @@
...
 
-## Useful links (Optional)
-> Provide links to the most relevant module documentation (tutorials, technical references, resources, etc.).
+## Features
 
-## Feedback (Optional)
-> Describe how users can provide feedback.
\ No newline at end of file
+Cloud Manager can provision following cloud resources in the underlying cloud provider subscription:
```
Comment (mmitoraj) [inline docs/user/README.md:12]:
```suggestion
Cloud Manager can provision the following cloud resources in the underlying cloud provider subscription:
```

File: docs/user/README.md [hunk 17]
```diff
@@ -1,14 +1,45 @@
...
+* NFS server that can be used as ReadWriteMany volume in the Kyma cluster
```
Comment (mmitoraj) [inline docs/user/README.md:13]:
```suggestion
* NFS server that can be used as a ReadWriteMany (RWX) volume in the Kyma cluster
```

File: docs/user/README.md [hunk 18]
```diff
@@ -1,14 +1,45 @@
...
+
+## Architecture
+Kyma Cloud Manager Operator runs in the Kyma Control Plane and does remote reconciliation on Kyma clusters that
+have `cloud-manager` module enabled. It brings various Custom Resource Definitions each representing some
```
Comment (mmitoraj) [inline docs/user/README.md:17]:
```suggestion
have the Cloud Manager module enabled. It brings various Custom Resource Definitions (CRDs) each representing some
```

File: docs/user/README.md [hunk 19]
```diff
@@ -1,14 +1,45 @@
...
+specific cloud resource from the underlying cloud provider subscription.
```
Comment (mmitoraj) [inline docs/user/README.md:18]:
```suggestion
a specific cloud resource from the underlying cloud provider subscription. 
```

File: docs/user/README.md [hunk 20]
```diff
@@ -1,14 +1,45 @@
...
+## Architecture
+Kyma Cloud Manager Operator runs in the Kyma Control Plane and does remote reconciliation on Kyma clusters that
+have `cloud-manager` module enabled. It brings various Custom Resource Definitions each representing some
+specific cloud resource from the underlying cloud provider subscription. 
+
+## API / Custom Resources Definitions
+
+### IpRange CR
+
+The `iprange.cloud-resources.kyma-project.io` CustomResourceDefinition (CRD) describes the VPC network
```
Comment (mmitoraj) [inline docs/user/README.md:24]:
```suggestion
The `iprange.cloud-resources.kyma-project.io` CRD describes the VPC network
```

File: docs/user/README.md [hunk 21]
```diff
@@ -1,14 +1,45 @@
...
+IP range that will be used for IP address allocation for cloud resources that require IP address.
```
Comment (mmitoraj) [inline docs/user/README.md:25]:
```suggestion
IP range used for IP address allocation for cloud resources that require an IP address.
```

File: docs/user/README.md [hunk 22]
```diff
@@ -1,14 +1,45 @@
...
+To learn more, read the [IpRange custom resource documentation](./resources/04-10-iprange.md).
```
Comment (mmitoraj) [inline docs/user/README.md:26]:
```suggestion
To learn more, read the [IpRange Custom Resource](./resources/04-10-iprange.md) documentation.
```

File: docs/user/README.md [hunk 23]
```diff
@@ -1,14 +1,45 @@
...
+
+
+### AwsNfsVolume CR
+
+The `awsnfsvolume.cloud-resources.kyma-project.io` CustomResourceDefinition (CRD) describes the AWS EFS
```
Comment (mmitoraj) [inline docs/user/README.md:31]:
```suggestion
The `awsnfsvolume.cloud-resources.kyma-project.io` CRD describes the AWS EFS
```

File: docs/user/README.md [hunk 24]
```diff
@@ -1,14 +1,45 @@
...
+instance that can be used as RWX volume in the cluster.
+To learn more, read the [AwsNfsVolume custom resource documentation](./resources/04-20-10-aws-nfs-volume.md).
```
Comment (mmitoraj) [inline docs/user/README.md:33]:
```suggestion
To learn more, read the [AwsNfsVolume Custom Resource](./resources/04-20-10-aws-nfs-volume.md) documentation.
```

File: docs/user/README.md [hunk 25]
```diff
@@ -1,14 +1,45 @@
...
+
+
+### GcpNfsVolume CR
+
+The `gcpnfsvolume.cloud-resources.kyma-project.io` CustomResourceDefinition (CRD) describes the GCP Filestore
```
Comment (mmitoraj) [inline docs/user/README.md:38]:
```suggestion
The `gcpnfsvolume.cloud-resources.kyma-project.io` CRD describes the GCP Filestore
```

File: docs/user/README.md [hunk 26]
```diff
@@ -1,14 +1,45 @@
...
+instance that can be used as RWX volume in the cluster.
+To learn more, read the [GcpNfsVolume custom resource documentation](./resources/04-30-10-gcp-nfs-volume.md).
```
Comment (mmitoraj) [inline docs/user/README.md:40]:
```suggestion
To learn more, read the [GcpNfsVolume Custom Resource](./resources/04-30-10-gcp-nfs-volume.md)  documentation.
```

File: docs/user/tutorials/01-10-aws-nfs-volume.md [hunk 27]
```diff
@@ -0,0 +1,188 @@
...
+    > skip this step, and do not delete the default IpRange.
+
+    ```shell
+    kubectl delete -n kyma-system iprange default
+    ```
+
+  * Remove the created namespace:
+    ```shell
+    kubectl delete namespace $NAMESPACE
+    ```
```
Comment (mmitoraj) [inline docs/user/tutorials/01-10-aws-nfs-volume.md:188]: We could indent those bullet points so that they show up under step 8.

**Review by mmitoraj — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by vandjelk — APPROVED** [review]

**Pull request merged**